hack/istio/install-testing-demos.sh

```diff
@@ -152,6 +191,7 @@ else
 
   echo "Deleting the 'sleep' app in the 'sleep' namespace..."
   ${CLIENT_EXE} delete -n sleep -f ${ISTIO_DIR}/samples/sleep/sleep.yaml
+  ${CLIENT_EXE} delete smm default -n "sleep" --ignore-not-found=true
```

Comment:
Because line 10 has `set -eu` - this line is going to cause this script to abort right here on a non-Maistra environment. The `--ignore-not-found=true` doesn't help - this will still exit with non-zero if the type ServiceMeshMember doesn't exist:

```
$ kubectl delete smm default -n "sleep" --ignore-not-found=true
error: the server doesn't have a resource type "smm"
$ echo $?
1
```
That means it will never get to the lines below this and thus fail to clean up non-Maistra environments.
You should wrap this around IS_MAISTRA if-stmt so this only gets calls in Maistra environments.